docs/guide/installation-guide.txt

```diff
@@ -1,3 +1,128 @@
 ==================
 Installation Guide
 ==================
+
+The recommended way to get started using the Node.js driver is by using ``NPM`` (Node Package Manager) to install the dependency in your project.
+
+MongoDB Driver
+--------------
+
+After you've created your project with ``npm init`` , you can install the MongoDB driver and its dependencies with the command:
+
+.. code-block:: sh
+
+   npm install mongodb --save
+
+This will download the MongoDB driver and add a dependency entry in your ``package.json`` file.
+
+Troubleshooting
+---------------
+
+The MongoDB driver depends on several other packages, including:
+
+* `bson <https://www.npmjs.com/package/bson>`_
+* `require_optional <https://www.npmjs.com/package/require_optional>`_
+* `safe-buffer <https://www.npmjs.com/package/safe-buffer>`_
+* `saslprep <https://www.npmjs.com/package/saslprep>`_
+
+Additionally, there are multiple optional dependencies that can be installed alongside the driver:
+
+* `bson-ext <https://www.npmjs.com/package/bson-ext>`_
+* `kerberos <https://www.npmjs.com/package/kerberos>`_
+* `mongodb-client-encryption <https://www.npmjs.com/package/mongodb-client-encryption>`_
+* `snappy <https://www.npmjs.com/package/snappy>`_
+
+These optional modules are all native C++ extensions. They are optional extensions and are not required for the driver to function.
+
+bson-ext Module
+---------------
+
+The ``bson-ext`` module is an alternative **BSON** parser that is written in C++. If you wish to use the ``bson-ext`` module you will need to add the ``bson-ext`` module to your module's dependencies.
+
+.. code-block:: sh
+
+   npm install bson-ext --save
+
```

Comment:
I recommend adding a section about "prebuild"s for our binary packages. Say that we attempt to use prebuild for a majority of our C++ extensions, so people probably won't need a local compiler.

Reply:
FWIW I don't think we have prebuild for bson-ext

Reply:
ya I was more pointing out adding it at this place in the docs, maybe it should be before any of the sections about the modules that might have prebuilds. Also, in the future when all of these are moved to EVG they should have prebuilds